### PR TITLE
feat: append sub nav url and redirect to contributions if there is no highlights

### LINKED ISSUE #914: Feature: User profiles sub nav should append the URL
### Type of feature

🍕 Feature

### Current behavior

When viewing a user profile, I click on highlights and contributions but the URL does not update. 

Screen Shot 2023-02-27 at 12 24 56 PM

### Suggested solution

Screen Shot 2023-02-27 at 12 24 56 PM

examples URL: 
```
https://insights.opensauced.pizza/user/bdougie/contributions
https://insights.opensauced.pizza/user/bdougie/highlights
```

### Additional context

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### Contributing Docs

- [ ] I agree to follow this project's Contribution Docs

### LINKED ISSUE #916: Feature: Default to contributions if no highlights exists for the viewer of the profile.
### Type of feature

🍕 Feature

### Current behavior

Screen Shot 2023-02-27 at 6 25 07 PM

Users are seeing blank pages, but we should default to contributions when no highlights exist.

### Suggested solution

1. When the viewer is not the profile being viewed. Check to see if highlights exist, if not, show contributions by default. 

### Additional context

related https://github.com/open-sauced/insights/issues/914

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### Contributing Docs

- [ ] I agree to follow this project's Contribution Docs

### PR DESCRIPTION
**Pull request by TAKANOME-DEV**
## What type of PR is this? (check all applicable)

- [x] 🍕 Feature
- [ ] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI
- [ ] 📦 Chore (Release)
- [ ] ⏩ Revert

## Description

This PR:
- redirects to the contributions tab if the user has no highlights by default (check the [comment below](https://github.com/open-sauced/insights/pull/918#issuecomment-1450285828) for more details)
- adds `/user/[username]/highlights.tsx` page to show user's highlights
- adds `/user/[username]/contributions.tsx` page to show user's contributions
## Related Tickets & Documents

Fixes #914 
Fixes #916
## Mobile & Desktop Screenshots/Recordings

https://user-images.githubusercontent.com/79809121/221815102-d84f7ec0-4897-4e43-99cb-278a291a5b42.mp4

## Added tests?

- [ ] 👍 yes
- [x] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help

## Added to documentation?

- [ ] 📜 README.md
- [ ] 📓 docs.opensauced.pizza
- [ ] 🍕 dev.to/opensauced
- [ ] 📕 storybook
- [x] 🙅 no documentation needed

### CODE REVIEW

**Comment by OgDev-01** [comment]
Just tested and I was taken to the contributor's tab even when I have highlights.

### recording of the test
https://www.loom.com/share/83bacaeb233640378f060ac9a2016088

File: components/organisms/ContributorProfileTab/contributor-profile-tab.tsx [hunk 1]
```diff
@@ -44,22 +45,32 @@ const ContributorProfileTab = ({
...
+  const hasHighlights = highlights?.length > 0;
+  const router = useRouter();
+
+  const handleTabUrl = (tab: string) => {
+    router.push(`/user/${login}/${tab.toLowerCase()}`);
+  };
+
   useEffect(() => {
     setInputVisible(highlights && highlights.length !== 0 ? true : false);
+    if (!hasHighlights) router.push(`/user/${login}/contributions`);
```
Comment (OgDev-01) [inline components/organisms/ContributorProfileTab/contributor-profile-tab.tsx:57]: we should add an else statement here to  push to `/user/${login}/highlights`

Comment (TAKANOME-DEV) [reply components/organisms/ContributorProfileTab/contributor-profile-tab.tsx:57]: Absolutely, thanks 👍

**Comment by brandonroberts** [comment]
@TAKANOME-DEV How can we restrict valid routes for the `username`? Renaming `[username].tsx` to `[...username].tsx` makes it capture anything after `/user` which allows you to type invalid pages such as `/user/{username}/highlight`.

In the future, we may have a URL for an individual user highlight, and this makes that more difficult.

Granted, nested routing would make this much more straightforward, but that's another effort altogether.

**Comment by TAKANOME-DEV** [comment]
> @TAKANOME-DEV How can we restrict valid routes for the `username`? Renaming `[username].tsx` to `[...username].tsx` makes it capture anything after `/user` which allows you to type invalid pages such as `/user/{username}/highlight`.
> 
> In the future, we may have a URL for an individual user highlight, and this makes that more difficult.
> 
> Granted, nested routing would make this much more straightforward, but that's another effort altogether.

I didn't think about that, thanks :+1: 

I wanted to suggest using `user/[username]?tab=[currentTab]` :sweat: That's what I did in my first commit https://github.com/open-sauced/insights/pull/918/commits/2ac9cf977753d8eeacacef7879a72dea47b244a6

@brandonroberts do you have any suggestions?

**Review by bdougie — CHANGES_REQUESTED** [review]
When I visit this [link](https://deploy-preview-918--oss-insights.netlify.app/user/bdougie/contributions) I'd expect to see contributions, but it redirects to /highlights instead.

https://deploy-preview-918--oss-insights.netlify.app/user/bdougie/contributions

**Comment by brandonroberts** [comment]
> 

@TAKANOME-DEV I don't know if this is a pattern or not, but one option would be to import/re-export the contributor profile page under `[username]/highlights.tsx` and `[username]/contributions.tsx`.

```ts
// pages/user/[username]/highlights.tsx
import ContributorProfilePage from "../[username]";

export default ContributorProfilePage;
```

From there we can determine which page you're on from in the page component using the `router.pathname`.

Thoughts?

**Comment by TAKANOME-DEV** [comment]
>I don't know if this is a pattern or not, but one option would be to import/re-export the contributor profile page under `[username]/highlights.tsx` and `[username]/contributions.tsx`.
> 
Good option :+1: 
> Subjectively, I think having it be part of the path instead of query params is better for linking.
> 
Cool, will go with paths then :+1:

**Comment by brandonroberts** [comment]
@TAKANOME-DEV I tried visiting [https://deploy-preview-918--oss-insights.netlify.app/user/0-vortex/contributions an](https://deploy-preview-918--oss-insights.netlify.app/user/0-vortex/highlights) and it took me to `/contributions` but still shows the highlights tab

**Comment by TAKANOME-DEV** [comment]
> @TAKANOME-DEV I tried visiting [https://deploy-preview-918--oss-insights.netlify.app/user/0-vortex/contributions an](https://deploy-preview-918--oss-insights.netlify.app/user/0-vortex/highlights) and it took me to `/contributions` but still shows the highlights tab
>
Since that user does not have any highlights, it redirects to the contributions by default but the current tab did not update.
It should be fixed now :+1:

**Review by bdougie — APPROVED** [review]
Works as expected now

**Review by OgDev-01 — APPROVED** [review]
LGTM

**Comment by 0-vortex** [comment]
> > @TAKANOME-DEV I tried visiting [https://deploy-preview-918--oss-insights.netlify.app/user/0-vortex/contributions an](https://deploy-preview-918--oss-insights.netlify.app/user/0-vortex/highlights) and it took me to `/contributions` but still shows the highlights tab
> 
> Since that user does not have any highlights, it redirects to the contributions by default but the current tab did not update. It should be fixed now 👍

It's a 3 redirect loop on my end, we might have to push the alpha router/await fixes to stop the page from FOUTing! 🍕

**Comment by brandonroberts** [comment]
@TAKANOME-DEV still not seeing the redirect on my end

https://user-images.githubusercontent.com/42211/222166293-c14da4c1-b458-4e8a-996b-f4d664bdbcc9.mov

**Comment by brandonroberts** [comment]
@TAKANOME-DEV thanks for the explanation. That makes more sense to me now

**Pull request merged**